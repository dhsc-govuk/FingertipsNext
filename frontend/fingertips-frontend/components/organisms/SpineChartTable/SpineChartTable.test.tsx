import { render, screen } from '@testing-library/react';
import { expect } from '@jest/globals';
import { SpineChartTableHeadingEnum, SpineChartTableHeader } from '.';

describe('Spine chart table suite', () => {
  const mockHeaderData = {
    area: 'testArea',
    group: 'testGroup'}

  const EmptySpan = 3
  
  describe('Spine chart table header', () => {

    it('should contain the expected elements', () => {
      render(
        <SpineChartTableHeader
          areaName={mockHeaderData.area}
          groupName={mockHeaderData.group}
        />
      );

      expect(screen.getByTestId('empty-header')).toHaveTextContent(
        ''
      );
      expect(screen.getByTestId('area-header')).toHaveTextContent(
        `${mockHeaderData.area}`
      );
      expect(screen.getByTestId('group-header')).toHaveTextContent(
        `${mockHeaderData.group}`
      );
      expect(screen.getByTestId('england-header')).toHaveTextContent(
        `Benchmark: England`
      );

      Object.values(SpineChartTableHeadingEnum).forEach((heading, index) =>
        expect(
          screen.getByTestId(`${heading}-header-${index}`)
        ).toBeInTheDocument()
      );
    });
  });
});
