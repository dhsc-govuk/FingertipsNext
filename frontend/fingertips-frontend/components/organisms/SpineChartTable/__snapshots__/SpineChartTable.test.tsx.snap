// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spine chart table suite Spine chart table snapshot test - should match snapshot 1`] = `
<DocumentFragment>
  .c20 {
  margin: 0;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  margin-bottom: 15px;
}

.c20 > p {
  margin: 0;
}

.c20 > p > code {
  padding: 0.2em 0.4em;
  margin: 0;
  font-size: 85%;
  background-color: rgba(27,31,35,0.05);
  border-radius: 3px;
}

.c20 > pre {
  padding: 16px;
  overflow: auto;
  font-size: 85%;
  line-height: 1.45;
  background-color: #f6f8fa;
  border-radius: 3px;
}

.c20 > pre > code {
  display: inline;
  padding: 0;
  margin: 0;
  border: 0;
  overflow: visible;
  line-height: inherit;
  word-wrap: normal;
}

.c17 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 14px;
  line-height: 1;
  display: inline-block;
  padding-top: 5px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background-color: #1d70b8;
  color: #ffffff;
}

.c3 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
  font-weight: 700;
}

.c3:last-child {
  padding-right: 0;
}

.c13 {
  padding: 10px 20px 10px 0;
  border-bottom: 1px solid #bfc1c3;
  text-align: left;
}

.c13:last-child {
  padding-right: 0;
}

.c1 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 16px;
  line-height: 1.25;
  color: #0b0c0c;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
  margin-bottom: 20px;
}

.c4 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  font-weight: bold;
  padding: 0.625em 0;
}

.c6 {
  text-align: right;
  padding-right: 10px;
  vertical-align: top;
}

.c7 {
  background-color: #B1B4B6;
  border-top: solid #F3F2F1 2px;
  vertical-align: top;
}

.c14 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 14px;
  line-height: 1.1428571428571428;
  padding-right: 0;
}

.c22 {
  text-align: right;
  padding-right: 10px;
}

.c15 {
  text-align: left;
  width: 10%;
}

.c5 {
  text-align: left;
  vertical-align: top;
}

.c10 {
  text-align: center;
}

.c16 {
  text-align: center;
}

.c8 {
  background-color: #F3F2F1;
  border-top: #B1B4B6;
  text-align: left;
}

.c11 {
  background-color: #F3F2F1;
  border-top: #B1B4B6;
  text-align: right;
}

.c9 {
  background-color: #B1B4B6;
  border-top: #F3F2F1;
  text-align: center;
}

.c12 {
  background-color: #B1B4B6;
  border-top: #F3F2F1;
  text-align: right;
}

.c23 {
  background-color: #F3F2F1;
  border-top: #B1B4B6;
  text-align: right;
}

.c24 {
  background-color: #B1B4B6;
  border-top: #F3F2F1;
  text-align: right;
}

.c19 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c18 {
  text-transform: unset;
  margin: 0.3125em;
  color: #303436;
  background-color: #E1E2E3;
}

.c25 {
  text-transform: unset;
  margin: 0.3125em;
  color: #270D0B;
  background-color: #F6C6BF;
}

.c21 {
  margin-bottom: 0;
  margin-left: 0.3125em;
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 400;
  font-size: 12px;
  line-height: 1.25;
}

.c0 {
  overflow-x: scroll;
}

.c2 {
  display: block;
  width: 100%;
  table-layout: fixed;
}

@media print {
  .c20 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c20 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media only screen and (min-width:641px) {
  .c20 {
    margin-bottom: 20px;
  }
}

@media print {
  .c17 {
    font-size: 14px;
    line-height: 1;
  }
}

@media only screen and (min-width:641px) {
  .c17 {
    font-size: 16px;
    line-height: 1;
  }
}

@media print {
  .c1 {
    font-size: 14px;
    line-height: 1.15;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    font-size: 19px;
    line-height: 1.3157894736842106;
  }
}

@media print {
  .c1 {
    color: #000;
  }
}

@media only screen and (min-width:641px) {
  .c1 {
    margin-bottom: 30px;
  }
}

@media print {
  .c4 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c4 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media print {
  .c14 {
    font-size: 14px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c14 {
    font-size: 16px;
    line-height: 1.25;
  }
}

@media print {
  .c21 {
    font-size: 12px;
    line-height: 1.2;
  }
}

@media only screen and (min-width:641px) {
  .c21 {
    font-size: 14px;
    line-height: 1.4285714285714286;
  }
}

<div
    class="c0"
    data-testid="spineChartTable-component"
  >
    <table
      class="c1 c2"
    >
      <tbody
        class=""
      >
        <tr
          class=""
        >
          <th
            class="c3"
            colspan="3"
            data-testid="empty-header"
          />
          <th
            class="c3 c4 c5"
            colspan="3"
            data-testid="area-header"
          >
            Greater Manchester ICB - 00T
          </th>
          <th
            class="c3 c4 c6 c7 c8"
            data-testid="group-header"
          >
            Manchester
          </th>
          <th
            class="c3 c4 c6 c7 c9"
            colspan="3"
            data-testid="england-header"
          >
            Benchmark: England
          </th>
        </tr>
        <tr
          class=""
        >
          <th
            class="c3 c4 c5"
            data-testid="Indicator-header"
          >
            Indicator
          </th>
          <th
            class="c3 c4 c5"
            data-testid="Unit-header"
          >
            Unit
          </th>
          <th
            class="c3 c4 c5 c10"
            data-testid="Period-header"
          >
            Period
          </th>
          <th
            class="c3 c4 c5 c10"
            data-testid="Recent trend-header"
          >
            Recent trend
          </th>
          <th
            class="c3 c4 c5 c10"
            data-testid="Count-header"
          >
            Count
          </th>
          <th
            class="c3 c4 c6"
            data-testid="AreaValue-header"
          >
            Value
          </th>
          <th
            class="c3 c4 c6 c7 c11"
            data-testid="GroupValue-header"
          >
            Value
          </th>
          <th
            class="c3 c4 c6 c7 c12"
            data-testid="Value-header"
          >
            Value
          </th>
          <th
            class="c3 c4 c6 c7 c12"
            data-testid="Worst-header"
          >
            Worst
          </th>
          <th
            class="c3 c4 c6 c7 c12"
            data-testid="Range-header"
          >
            Range
          </th>
          <th
            class="c3 c4 c6 c7 c12"
            data-testid="Best-header"
          >
            Best
          </th>
        </tr>
        <tr
          class=""
        >
          <td
            class="c13 c14 c15"
            data-testid="indicator-cell"
          >
            Test indicator 2
          </td>
          <td
            class="c13 c14 c15"
            data-testid="unit-cell"
          >
            per 1,000
          </td>
          <td
            class="c13 c14 c15 c16"
            data-testid="period-cell"
          >
            2024
          </td>
          <td
            class="c13 c14 c15 c16"
          >
            <div
              data-testid="trendTag-container"
            >
              <strong
                class="c17 c18"
              >
                <div
                  class="c19"
                  data-testid="trend-tag-component"
                >
                  <div>
                    <div
                      class="c20 c21"
                    >
                      <p>
                        No trend data available
                      </p>
                    </div>
                  </div>
                </div>
              </strong>
            </div>
          </td>
          <td
            class="c13 c14 c15 c16"
            data-testid="count-cell"
          >
            111
          </td>
          <td
            class="c13 c14 c22"
            data-testid="value-cell"
          >
            690.3
          </td>
          <td
            class="c13 c14 c22 c23"
            data-testid="group-value-cell"
          >
            690.3
          </td>
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-value-cell"
          >
            135.1
          </td>
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-worst-cell"
          >
            100.0
          </td>
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-range"
          />
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-best-cell"
          >
            22.0
          </td>
        </tr>
        <tr
          class=""
        >
          <td
            class="c13 c14 c15"
            data-testid="indicator-cell"
          >
            Test indicator 1
          </td>
          <td
            class="c13 c14 c15"
            data-testid="unit-cell"
          >
            kg
          </td>
          <td
            class="c13 c14 c15 c16"
            data-testid="period-cell"
          >
            2008
          </td>
          <td
            class="c13 c14 c15 c16"
          >
            <div
              data-testid="trendTag-container"
            >
              <strong
                class="c17 c25"
              >
                <div
                  class="c19"
                  data-testid="trend-tag-component"
                >
                  <div>
                    <svg
                      data-testid="arrow-icon"
                      fill="none"
                      height="15"
                      stroke="#270D0B"
                      stroke-linecap="square"
                      stroke-width="1.5"
                      viewBox="0 0 20 20"
                      width="15"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12 19V6M5 12l7-7 7 7"
                        data-testid="arrow-up"
                      />
                    </svg>
                  </div>
                  <div>
                    <div
                      class="c20 c21"
                    >
                      <p>
                        Increasing and getting worse
                      </p>
                    </div>
                  </div>
                </div>
              </strong>
            </div>
          </td>
          <td
            class="c13 c14 c15 c16"
            data-testid="count-cell"
          >
            222
          </td>
          <td
            class="c13 c14 c22"
            data-testid="value-cell"
          >
            890.3
          </td>
          <td
            class="c13 c14 c22 c23"
            data-testid="group-value-cell"
          >
            980.3
          </td>
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-value-cell"
          >
            890.3
          </td>
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-worst-cell"
          >
            959.0
          </td>
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-range"
          />
          <td
            class="c13 c14 c22 c24"
            data-testid="benchmark-best-cell"
          >
            1,666.0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
